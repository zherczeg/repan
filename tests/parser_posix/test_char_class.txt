# Test character sets

/\?\[[\a-z][]-]/X
    print

/[\-\]/X
    print

/[c-a]/X
    print

/[@#/X
    print

# Test character ranges

/[[:upper:]-]+?/X
    print

/[---[:alpha:]-a]/X
    print
