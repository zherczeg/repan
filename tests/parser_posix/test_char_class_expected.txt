# Test character sets

/\?\[[\a-z][]-]/X
    print
    /\?\[[\\a-z][\]\-]/

/[\-\]/X
    print
    /[\\]/

/[c-a]/X
    print
    Pattern parse failed: Range start and end are out of order
        /[[ HERE ]>>c-a]/

/[@#/X
    print
    Pattern parse failed: Unterminated [...] character class
        /[ HERE ]>>[@#/

# Test character ranges

/[[:upper:]-]+?/X
    print
    /(?:[[:upper:]\-]+)?/

/[---[:alpha:]-a]/X
    print
    Pattern parse failed: Neither the start nor the end of a range can be a character class
        /[---[ HERE ]>>[:alpha:]-a]/
